lib/workers/repository/update/pr/changelog/source-github.ts

```diff
@@ -121,7 +121,7 @@ export async function getChangeLogJSON(
 
   const cacheNamespace = 'changelog-github-release';
   function getCacheKey(prev: string, next: string): string {
-    return `${manager}:${depName}:${prev}:${next}`;
+    return `${sourceUrl}:${depName}:${prev}:${next}`;
```

Comment:
We need to slugify this or some other form of massaging to be sure that it's a valid cache key

Reply:
actually the previous `repositrory/pathname` was exactly that, it was referring to the `sourceUrl` and not the repositroy renovate is ran on.

Reply:
for example
if `sourceUrl =  https://github.com/sindresorhus/delay`
then 
for github `repositroy=sindresorhus/delay` and pathnam is the same for gitlab

Reply:
I still prefer to slugify it to be safe

Reply:
so i tried [slugify](https://www.npmjs.com/package/slugify) as it is already imported and used. 
Using it with URLs causes slashes to be removed.
this means that in the following edge case we will end up with corrupted data:
for a given dep, we have two sourceUrls:
1) `http://domain/org/repo`
2) `http://domain/orgrep/o`

both will translate to the same key:

`http:domainorgrepo:depName:fromVer:toVer`

wdyt?

Reply:
i think ill just replace slashes with spaces before using it.

Reply:
I had expected it to replace / with -. Considering all the other workarounds we have for it, we should probably write our own slugify function to replace it eventually